src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -234,6 +246,19 @@ private static void LogErrors(ScriptHost host)
             }
         }
 
+        /// <summary>
+        /// Handler for any file changes under the root script path
+        /// </summary>
+        private void OnScriptFileChanged(object sender, FileSystemEventArgs e)
+        {
+            if (_restartDelayTokenSource != null && !_restartDelayTokenSource.IsCancellationRequested)
+            {
+                // if we're currently awaiting an error/restart delay
```

Comment:
The theory is that any host errors are due to invalid user files. If you haven't touched them in a while, we'll backoff. When you finally go fix the issue, we restart immediately. I was also considering weaving IsDebug mode into this, but that might complicate things unecessarily.